inject/src/main/java/io/avaje/inject/spi/GenericTypeUtil.java

```diff
@@ -56,7 +54,323 @@ static Type typeArgument(Class<?> clazz) {
       // needed type argument not supplied - break and throw exception
       break;
     }
-    throw new IllegalArgumentException(currentType + " does not specify the type parameter T of GenericType<T>");
+    throw new IllegalArgumentException(
+        currentType + " does not specify the type parameter T of GenericType<T>");
+  }
+
+  /** Returns an array type whose elements are all instances of {@code componentType}. */
+  public static GenericArrayType arrayOf(Type elementType) {
```

Comment:
Looks like this has come in from avaje-jsonb.  Why is is added though?  Not part of bug fix per say.